sherpa/instrument.py

```diff
@@ -413,102 +468,231 @@ def _create_tail_grid(axis_list):
 class PSFModel(Model):
     """Convolve a model by another model or data set.
 
+    At the moment the code does not distinguish between 1D and 2D data
+    and models.
+
+    Parameters
+    ----------
+    name : str
+        The name for the model.
+    kernel : sherpa.data.Data instance, callable, or None, optional
+        The kernel used to convolve models. This can be changed.
+
     Notes
     -----
     A number of attributes are displayed as parameters, if set, but
     are not handled as parameters. The attributes are: kernel, size,
-    centre, and origin.
+    origin, and center. The size, center, and origin values are only
+    displayed when set (and will be set by the `fold` method if
+    needed). There is an attempt to ensure that the size, origin, and
+    center fields have the correct size - that is they match the
+    dimensionality of the kernel - but it is possible for an invalid
+    combination to be set.
 
     """
 
     def __init__(self, name='psfmodel', kernel=None):
+
+        # store the name without the leading "psfmodel." term that Model adds.
         self._name = name
+
         self._size = None
         self._origin = None
         self._center = None
         self._must_rebin = False
+        self._model = None
+
+        self._kernel = None
```

Comment:
I am reluctant to make what-appear-to-be-simple clean-up changes in the instrument code as "it is complicated" and I don't feel we have a good-enough set of tests that cover the desired functionality (I've been adding tests but these are to check certain code paths, rather than being a meaningful set of tests that cover how users are expected to use these).

I note that a number of these fields are internal to the class, as they are used to populate properties (e.g. "size", "origin", "center") so the desire for a docstring is, I think, somewhat lessened from if these are user-visible fields.

I would expect any subclass to call this routine, but I can't guarantee it.